src/EventHandler/Message/Private/SecretMessage.php

```diff
@@ -0,0 +1,76 @@
+<?php declare(strict_types=1);
+
+/**
+ * This file is part of MadelineProto.
+ * MadelineProto is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+ * MadelineProto is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ * You should have received a copy of the GNU General Public License along with MadelineProto.
+ * If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @author    Mahdi <mahdi.talaee1379@gmail.com>
+ * @copyright 2016-2023 Mahdi <mahdi.talaee1379@gmail.com>
+ * @license   https://opensource.org/licenses/AGPL-3.0 AGPLv3
+ * @link https://docs.madelineproto.xyz MadelineProto documentation
+ */
+
+namespace danog\MadelineProto\EventHandler\Message\Private;
+
+use danog\MadelineProto\EventHandler\Media\Audio;
+use danog\MadelineProto\EventHandler\Media\Decrypted as SecretMedias;
+use danog\MadelineProto\EventHandler\Media\Document;
+use danog\MadelineProto\EventHandler\Media\DocumentPhoto;
+use danog\MadelineProto\EventHandler\Media\Gif;
+use danog\MadelineProto\EventHandler\Media\MaskSticker;
+use danog\MadelineProto\EventHandler\Media\Photo;
+use danog\MadelineProto\EventHandler\Media\RoundVideo;
+use danog\MadelineProto\EventHandler\Media\Sticker;
+use danog\MadelineProto\EventHandler\Media\Video;
+use danog\MadelineProto\EventHandler\Media\Voice;
+use danog\MadelineProto\EventHandler\Message\Entities\MessageEntity;
+use danog\MadelineProto\MTProto;
+
+final class SecretMessage extends AbstractPrivateMessage
+{
+    /** Content of the message */
```

Comment:
No need to duplicate these properties, we can just re-use the ones in privatemessage

Reply:
![Screenshot_20230906_160100](https://github.com/danog/MadelineProto/assets/73236713/2f7d1c71-32d1-4b3d-8366-10464538ed74)
As these properties is readonly i cant rewrite it directly with out duplicate it in SecretMessage class and as private chat id is different from secret chat id again i cant include it in PrivateMessage class!!

Reply:
No, these properties are already present in PrivateMessage, they can be populated directly by the constructor of PrivateMessage, there's no need to populate them in SecretMessage.
The `chatId` is also present in AbstractMessage, it's just slightly different for secret chats (and in this case, the logic to handle the distinction can also be present in AbstractMessage)